packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -177,12 +177,25 @@ function getInterfaceName(iface: Interface, options: TypeNameOptions | undefined
     interfaceName += `<${iface.templateMapper.args
       .map((x) => getTypeName(x, options))
       .join(", ")}>`;
+  } else if (iface.node.templateParameters.length > 0) {
```

Comment:
@timotheeguerin , I am not sure I have a good understanding of the template and template instance. it seems the latter is only triggered when an alias is used against a templated model/interface. It will be helpful if you can help to provide more insights about them. thanks.